### ISSUE TITLE
[bug] purity is too strict flagging console.logs

### ISSUE THREAD

**Issue by controversial** [Rel1cx/eslint-react#1616]
### Describe the bug

`purity` should have an option to ignore `console` calls within react component bodies, since this “side effect” is unobservable and broadly harmless, even if technically impure. 

### Reproduction

```tsx
export function MyComponent() {
  console.log('hi');
  return;
}
```

### Expected behavior

possible to disable `purity` from flagging console statements

### Platform and versions

```Text
3.0.0-rc.3
```
